pkg/acl/acl.go

```diff
@@ -330,6 +344,25 @@ func PrintACLsFromKafkaRestResponse(cmd *cobra.Command, acls []cckafkarestv3.Acl
 	return list.Print()
 }
 
+func principalHasIntegerId(principal string) (bool, error) {
+	if principal == "" {
+		return false, nil
+	}
+
+	x := strings.Split(principal, ":")
+	if len(x) < 2 {
+		return false, errors.Errorf("unrecognized principal format %s", principal)
+	}
+	suffix := x[1]
+
+	// The principal has a numeric ID
+	if _, err := strconv.ParseInt(suffix, 10, 32); err == nil {
```

Comment:
if this is the only condition for returning true, this function actually only needs two branches: this one, and the unrecognized error one. all other circumstances will return false.